File: c#/mybatisGen/WindowsFormsApplication1/bin/Debug/GenTarget/20190927212030/web/template/controller/HeartbeatEntityController.java
package web.template.controller;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import web.template.controller.common.BaseController;
import web.template.service.IRobotQrCodePayTaskService;
/**
 * ***模块的控制器类
 * @author admin
 */
@RestController
@RequestMapping("/HeartbeatEntity")
public class HeartbeatEntityController extends BaseController{
	
	@Autowired
	private HeartbeatEntityService service;
	
	/*抄考代码
	----------------------------------------------------------------------------------------------------------------------------
	HeartbeatEntity data = new HeartbeatEntity();

					data.setId(nextId());
		
					data.setLastHeartbeat(param.getLastHeartbeat());
		
		
		
					data.setRobotIp(param.getRobotIp());
		
		
					data.setRemark(param.getRemark());
		
		
					data.setExtendField(param.getExtendField());
		
		
					data.setMonitorServer(param.getMonitorServer());
		
		
	----------------------------------------------------------------------------------------------------------------------------
	HeartbeatEntityParams param = new HeartbeatEntityParams();

		param.setId(param.getId());
		
		param.setLastHeartbeat(param.getLastHeartbeat());
		
		param.setLastHeartbeatStart(param.getLastHeartbeatStart());
		
		param.setLastHeartbeatEnd(param.getLastHeartbeatEnd());
		
		param.setRobotIp(param.getRobotIp());
		
		param.setRobotIpLike(param.getRobotIpLike());
		
		param.setRemark(param.getRemark());
		
		param.setRemarkLike(param.getRemarkLike());
		
		param.setExtendField(param.getExtendField());
		
		param.setExtendFieldLike(param.getExtendFieldLike());
		
		param.setMonitorServer(param.getMonitorServer());
		
		param.setMonitorServerLike(param.getMonitorServerLike());
		
	----------------------------------------------------------------------------------------------------------------------------
	* 分页查询***模块，并返回查询结果
	* @param param	查询参数
    * @param currentPageIndex	当前页码
 	* @param pageSize	每页显示的数据量
    * @return	返回***模块的查询结果
    @RequestMapping("/page")
	public Result page(HeartbeatEntityParams param,int currentPageIndex,int pageSize){
		return new Result(0,null,service.page(param, currentPageIndex, pageSize));
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 导出***模块，使用excel表保存
	* @param param 筛选参数
	* @param excelType excel的后缀名、类型
	* @param currentPageIndex 当前页码
	* @param pageSize 每页显示的数据量
	* @return 返回excel
	@RequestMapping("/export")
	public ModelAndView export(HeartbeatEntityParams param, String excelType, int currentPageIndex, int pageSize){
		Map<String,Object> model=new HashMap<String,Object>();
		model.put(MyExcelView.EXCEL_DATA, service.export(param, currentPageIndex, pageSize).getPageDataList());
		model.put(MyExcelView.EXCEL_NAME, "excel文件名称."+excelType);
		model.put(MyExcelView.GROUP_NAME, "组名称");
		return new ModelAndView(myExcelView, model);
	}
	----------------------------------------------------------------------------------------------------------------------------
	@RequestMapping("/import")
	public Result importExcel(@RequestParam("fileUpload")MultipartFile fileUpload, 其他参数...){
		List<HeartbeatEntity> HeartbeatEntityList;
        if (fileUpload.getOriginalFilename().endsWith("xlsx"))
        {
            HeartbeatEntityList = ExcelHelper.excelXlsxToList<HeartbeatEntity>(fileUpload.InputStream);
        }
        else if (fileUpload.getOriginalFilename().endsWith("xls"))
        {
            HeartbeatEntityList = ExcelHelper.excelXlsToList<HeartbeatEntity>(fileUpload.InputStream);
        }
		service.addBatch(HeartbeatEntityList);
		return new Result(0, "导入成功。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	@RequestMapping("/del")
	public Result del(long id){
        service.del(id);
		return new Result(0, "数据已删除。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 新增一条数据
	* @param data	新增的数据
	@RequestMapping("/add")
	public Result add(HeartbeatEntity data){
		service.add(data);
		return new Result(0, "保存成功。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 批量修改状态
	* @param datas	  修改状态的数据
	@RequestMapping("/changeStatus")
	public Result changeStatus(HeartbeatEntity datas){
		return new Result(0,"修改成功，共"+service.changeStatus(datas)+"条。",null);
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 根据主键id查询***模块的数据实体
	* @param id	主键id
	* @param return 返回***模块的查询结果
	@RequestMapping("/load")
	public Result load(Long id){
		return new Result(0,null ,service.Load(id) );
	}
	----------------------------------------------------------------------------------------------------------------------------
	* 批量删除***模块的数据实体
	* @param datas	删除的数据主键列表
	@RequestMapping("/delBatch")
	public Result delBatch(List<HeartbeatEntity> datas){
		return new Result(0 ,null,"删除成功，共"+Service.DelBatch(datas)+"条。");
	}
	 */
}

File: c#/mybatisGen/WindowsFormsApplication1/bin/Debug/GenTarget/20190927084948/Controllers/HeartbeatEntityController.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Web;
using WebApplication1.Entity;
namespace WebApplication1.Controllers
{
	/// <summary>
	/// ***模块的控制器类
	/// </summary>
    public sealed partial class HeartbeatEntityController : BaseController
    {
		public HeartbeatEntityService Service { set; get; }

		/*抄考代码
----------------------------------------------------------------------------------------------------------------------------
		HeartbeatEntity data = new HeartbeatEntity
		{

					Id = Next(),
		
					LastHeartbeatTime = param.LastHeartbeatTime,
		
		
		
					RobotIp = param.RobotIp,
		
		
					Remark = param.Remark,
		
		
					ExtendField = param.ExtendField,
		
		
		};
----------------------------------------------------------------------------------------------------------------------------
		HeartbeatEntityParams param = new HeartbeatEntityParams
		{

			Id = param.Id,
		
			LastHeartbeatTime = param.LastHeartbeatTime,
		
			LastHeartbeatTimeStart = param.LastHeartbeatTimeStart,
		
			LastHeartbeatTimeEnd = param.LastHeartbeatTimeEnd,
		
			RobotIp = param.RobotIp,
		
			RobotIpLike = param.RobotIpLike,
		
			Remark = param.Remark,
		
			RemarkLike = param.RemarkLike,
		
			ExtendField = param.ExtendField,
		
			ExtendFieldLike = param.ExtendFieldLike,
		
		};
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 分页查询***模块，并返回查询结果
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的查询结果</returns>
		[Compress]
		public JsonResult Page(HeartbeatEntityParams param,int currentPageIndex = 1,int pageSize = 20)
		{
			return MyJson(new Result{code = 0, data = Service.Page(param, currentPageIndex, pageSize)});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导出***模块，使用excel表保存
        /// </summary>
        /// <param name="param">查询参数</param>
		/// <param name="excelType">excel类型</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		[OperInterval(IntervalMillisecond=10000)]
		[Compress]
		public ExcelResult<HeartbeatEntity> Export(HeartbeatEntityParams param, string excelType, int currentPageIndex = 1, int pageSize = 10000)
		{
			return new ExcelResult<HeartbeatEntity>
			{
				DataList = Service.Page(param, currentPageIndex, pageSize).pageDataList,
				FileName = "测试excel."+excelType
			};
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 导入***模块数据
        /// </summary>
        /// <param name="param">查询参数</param>
        /// <param name="currentPageIndex">当前页码</param>
        /// <param name="pageSize">每页显示的数据量</param>
        /// <returns>返回***模块的导出结果</returns>
		public JsonResult Import(HttpPostedFileBase fileUpload, 其他参数...)
		{
			List<HeartbeatEntity> HeartbeatEntityList;
            if (fileUpload.FileName.EndsWith("xlsx"))
            {
                HeartbeatEntityList = ExcelHelper.ExcelXlsxToList<HeartbeatEntity>(fileUpload.InputStream);
            }
            else if (fileUpload.FileName.EndsWith("xls"))
            {
                HeartbeatEntityList = ExcelHelper.ExcelXlsToList<HeartbeatEntity>(fileUpload.InputStream);
            }
			Service.AddBatch(HeartbeatEntityList);
			return MyJson(new Result { code = 0, msg = "导入成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键删除指定数据
        /// </summary>
        /// <param name="Id">删除数据的主键</param>
		public JsonResult Del(long Id)
		{
            Service.Del(Id);
			return MyJson(new Result { code = 0, msg = "数据已删除。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 新增一条数据
        /// </summary>
        /// <param name="data">新增的数据</param>
		public JsonResult Add(HeartbeatEntity data)
		{
			Service.Add(data);
			return MyJson(new Result { code = 0, msg = "保存成功。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量修改状态
        /// </summary>
        /// <param name="datas">修改状态的数据</param>
		public JsonResult ChangeStatus(HeartbeatEntity datas)
		{
			return MyJson(new Result { code = 0,msg=$"修改成功，共{Service.ChangeStatus(datas)}条。"});
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 根据主键id查询***模块的数据实体
        /// </summary>
        /// <param name="Id">主键id</param>
		/// <returns>返回***模块的查询结果</returns>
		public JsonResult Load(long Id)
		{
			return MyJson(new Result { code = 0, data = Service.Load(Id) });
		}
----------------------------------------------------------------------------------------------------------------------------
		/// <summary>
        /// 批量删除数据
        /// </summary>
        /// <param name="datas">批量删除的数据</param>
		public JsonResult DelBatch(List<HeartbeatEntity> datas)
		{
			return MyJson(new Result { code = 0 ,msg = $"删除成功，共{Service.DelBatch(datas)}条。" });
		}
		*/
    }
}